src/quality/Quality.tsx

```diff
@@ -115,11 +122,28 @@ export const Quality = ({
     [onSelectedRulesChange]
   )
 
-  const sampleColumns = useMemo(
-    () => getSampleColumns(columns, statistics?.column, rules, longestColumnsNames),
+  const referencedColumns = useMemo(
+    () => [...new Set(rules.map(rule => rule.referenced_column_names).flat())],
     [data]
   )
 
+  const _columns = useMemo(
```

Comment:
Is this not too much of "useMemo" blocks? Woulc it make sense to put those 3 blocks into one? It may be shorter, easier to read and consume less memory. Or not?